code/modules/overmap/ships/computers/comms.dm

```diff
@@ -0,0 +1,247 @@
+/obj/machinery/computer/ship/comms
+	name = "ship communication console"
+	icon_keyboard = "generic_key"
+	icon_screen = "helm"
+	light_color = "#7faaff"
+	core_skill = SKILL_DEVICES
+
+// TODO: add enabling or disabling comms transponder to hide transmission location
+// TODO: use comms console to target a specific entity with a specific maser rather than targetting everyone in range
+/obj/machinery/computer/ship/comms/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+
+	if(!linked)
+		display_reconnect_dialog(user, "Communication")
+		return
+
+	var/data[0]
+
+	var/list/printing_entities = list()
+	var/list/entities = list()
+	var/list/broadcasters = list()
+	var/list/receivers = list()
+
+	for(var/obj/machinery/shipcomms/comms in linked.comms_masers)
+		var/functional = !(comms.stat & (BROKEN|NOPOWER))
```

Comment:
This `!(stat & (BROKEN|NOPOWER))` and all the ones below should probably use operable()
Not sure what the general opinion on those stat proc calls are